File: leetcode/geek-time/linklist/func.go
package linklist

import "fmt"

type ListNode struct {
	Val  int
	Next *ListNode
}

func GetOneListNode(ii []int) *ListNode {
	p := &ListNode{
		Val:  ii[0],
		Next: nil,
	}
	f := p
	for i := 1; i < len(ii); i++ {
		f.Next = &ListNode{
			Val:  ii[i],
			Next: nil,
		}
		f = f.Next
	}
	return p
}

func Display(l *ListNode) {
	for l != nil {
		fmt.Println(l.Val)
		l = l.Next
	}
}

File: leetcode/geek-time/linklist/func.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


def arr2linklist(arr):
    root = ListNode(arr.pop(0))
    p = root
    for i in arr:
        p.next = ListNode(i)
        p = p.next
    return root


def iterate_linklist(head):
    print(head.val)
    while head.next is not None:
        head = head.next
        print(head.val)
